src/io.cpp

```
@@ -87,4 +93,107 @@ bool IO::ok()
+{
+    MapConfig mapData;
+    comm_->getMapConfig(mapData);
+    visualization_msgs::Marker pathMarker;
+    pathMarker.header.frame_id = "map";
+    pathMarker.header.stamp = ros::Time();
+    pathMarker.ns = "path";
+    pathMarker.id = id;
+    pathMarker.type = visualization_msgs::Marker::LINE_STRIP;
+    pathMarker.action = visualization_msgs::Marker::ADD;
+    pathMarker.color.a = 1.0;
+    pathMarker.color.r = color[0];
+    pathMarker.color.g = color[1];
+    pathMarker.color.b = color[2];
+    pathMarker.pose.orientation.w = 1.0;
+    pathMarker.scale.x = width;
+    for (std::vector<std::vector<double>>::iterator it = path.begin(); it != path.end(); ++it)
+    {
+        geometry_msgs::Point point;
+        if ((*it).size() < 2)
+        {
+            ROS_WARN_STREAM("point at index " << std::distance(path.begin(), it) << " is invalid, skipping");;
+            continue;
+        }
+        point.x = (*it)[0];
+        point.y = (*it)[1];
+        if ((*it).size() == 2)
+        {
+            point.z = 0;
+        }
+        else
+        {
+            point.z = (*it)[3];
+            if ((*it).size() > 3)
+            {
+                ROS_WARN_STREAM("point at index " << std::distance(path.begin(), it) << " has unused dimensions");
+            }
+        }
+
+        if (mapData.mapInitialised)
+        {
+            tf2::Transform tfPoint;
+            tfPoint.setOrigin(tf2::Vector3(point.x, point.y, point.z));
+            tfPoint.setRotation(tf2::Quaternion(0, 0, 0, 1));
+            tf2::Transform mapOffset;
+            mapOffset.setOrigin(tf2::Vector3(mapData.mapOffsetX, mapData.mapOffsetY, 0));
+            tf2::Quaternion q;
+            q.setRPY(0, 0, mapData.mapOrientation);
+            mapOffset.setRotation(q);
+
+            tfPoint = mapOffset * tfPoint;
+            point.x = tfPoint.getOrigin().x();
+            point.y = tfPoint.getOrigin().y();
+            point.z = tfPoint.getOrigin().z();
+        }
```

What if the mapdata is not initialised? Do you still publish in map frame

---

I'll rephrase. I think its better to throw an error and return false in this situation.